protocol/0009-MRKP-mark_price.md

```diff
@@ -6,70 +6,81 @@ A *Mark Price* is a concept derived from traditional markets.  It is a calculate
+### Flexible mark price methodology, combinations
 
-### 3. Oracle
+- It is possible to configure a cash settled futures market to use a weighted average of 1. weighted average of trades over `network.markPriceUpdateMaximumFrequency` and 2. impact of leveraged notional on the order book with the value of USDT `100` and 3. an oracle source and if last trade is last updated more than 1 minute ago then it is removed and the remaining re-weighted and if the oracle is last updated more than 5 minutes ago then it is removed and the remaining re-weighted (<a name="0009-MRKP-018" href="#0009-MRKP-018">0009-MRKP-018</a>) and a perps market (with the oracle source different to that used for the external price in the perps market) (<a name="0009-MRKP-019" href="#0009-MRKP-019">0009-MRKP-019</a>).
 
- An oracle source external to the market provides updates to the Mark Price. See the [data sourcing spec](./0045-DSRC-data_sourcing.md).
+- It is possible to configure a cash settled futures market to use a weighted average of 1. weighted average of trades over `network.markPriceUpdateMaximumFrequency` and 2. impact of leveraged notional on the order book with the value of USDT `100` and if last trade is last updated more than 1 minute ago then it is removed and the remaining re-weighted and if the oracle is last updated more than 5 minutes ago then it is removed and the remaining re-weighted and if both sources are stale than the mark price stops updating (<a name="0009-MRKP-020" href="#0009-MRKP-020">0009-MRKP-020</a>) and a perps market (<a name="0009-MRKP-021" href="#0009-MRKP-021">0009-MRKP-021</a>).
 
-### 4. Model
+- It is possible to configure a cash settled futures market to use a median of 1. weighted average of trades over `network.markPriceUpdateMaximumFrequency` and 2. impact of leveraged notional on the order book with the value of USDT `100` and 3. an oracle source and if last trade is last updated more than 1 minute ago then it is removed and if the oracle is last updated more than 5 minutes ago then it is removed (<a name="0009-MRKP-022" href="#0009-MRKP-022">0009-MRKP-022</a>) and a perps market (with the oracle source different to that used for the external price in the perps market) (<a name="0009-MRKP-023" href="#0009-MRKP-023">0009-MRKP-023</a>).
 
- The *Mark Price* may be calculated using a built in model.
+### Example 1 - A typical path of a cash settled futures market from end of openning auction till expiry (median mark price method)(<a name="0009-MRKP-024" href="#0009-MRKP-024">0009-MRKP-024</a>)
 
- >*Example:* An option price will theoretically decay with time even if the market's order book or trade history does not reflect this.
+1. Market is in opening auction, no mark price.
+2. Order uncrossed, ends of opening auction, market is in active state. New event is emitted for new mark price.
+3. New trade triggers new traded price, mark price recalculated, new event is emitted for new mark price.
```

Comment:
Mark price will only be recalculated when the time indicated by the mark price frequency is crossed, no?